README.md

```diff
@@ -378,6 +380,53 @@ sensu::agent::subscription { 'apache': }
 
 The resulting `agent.yml` would contain subscriptions for both `base` and `apache`.
 
+### Advanced agent - Annotations and Labels
+
+It's possible to define annotations and labels in many locations and the values merged into `agent.yml`:
+
+```puppet
+class { 'sensu::agent':
+  labels      => { 'foo' => 'bar', 'cpu.warning' => '70' },
+  annotations => { 'title.cpu' => 'foo' },
+}
+```
+
+Then in a profile class you can define the following:
+
+```puppet
+sensu::agent::label { 'cpu.warning': value => '90' }
+sensu::agent::annotation { 'summary.cpu': value => 'bar' }
+```
+
+The resulting `agent.yml` will contain the following:
+
+```yaml
+labels:
+  foo: 'bar'
+  cpu.warning: '90'
+annotations:
+  title.cpu: 'foo'
+  summary.cpu: 'bar'
+```
+
+**NOTE** `sensu::agent::annotation` and `sensu::agent::label` take precedence over values set by the class `sensu::agent`
```

Comment:
Does this also apply to subscriptions? If so, it should have this note as well.

Reply:
Not really, subscriptions are an array so the values are merged, so one doesn't take precedence over another.  I made a note that values are merged as it is sort of mentioned in the examples but not directly called out.